src/drivers/PortFolder/WDMTestingTemplate/driver/fail_driver1.c

```diff
@@ -0,0 +1,389 @@
+                            FALSE,                 
+                            &device                
+                            );
+    if(status==STATUS_SUCCESS)
+    {
+  
+       extension = (PDRIVER_DEVICE_EXTENSION)(device->DeviceExtension);
+
+       TopOfStack = IoAttachDeviceToDeviceStack (
+                                       device,
+                                       PhysicalDeviceObject);
+       
+       if (NULL == TopOfStack) 
+	   {
+           IoDeleteDevice(device);
+           return STATUS_DEVICE_REMOVED;
+       }
+
+
+       IoInitializeDpcRequest(device,DpcForIsrRoutine);
+
+       device->Flags &= ~DO_DEVICE_INITIALIZING;
+
+
+    }
+
+   return status;
+}
+
+_Use_decl_annotations_
+NTSTATUS
+DispatchCreate (
+    PDEVICE_OBJECT DeviceObject,
+    PIRP Irp
+    )
+{   
+    KAFFINITY ProcessorMask;
+    PDRIVER_DEVICE_EXTENSION extension ;    
+
+
+    PVOID *badPointer = NULL;
+
+    UNREFERENCED_PARAMETER(DeviceObject);
+    UNREFERENCED_PARAMETER(Irp);
+
+    PAGED_CODE();
+    //The check will mark the second occurance of PAGE_CODE as an error.
+    PAGED_CODE();
+
+    ExFreePool(badPointer);
+
+
+    extension = (PDRIVER_DEVICE_EXTENSION)DeviceObject -> DeviceExtension;
+
+    ProcessorMask   =  (KAFFINITY)1;
+    
+    IoConnectInterrupt( &extension->InterruptObject,
+                         InterruptServiceRoutine,
+                         extension,
+                         NULL,
+                         extension->ControllerVector,
+                         PASSIVE_LEVEL,
+                         PASSIVE_LEVEL,
+                         LevelSensitive,
+                         TRUE,
+                         ProcessorMask,
+                         TRUE );
+	NTSTATUS status;
+    status = Irp->IoStatus.Status = STATUS_SUCCESS;
+    IoCompleteRequest(Irp, IO_NO_INCREMENT);
+    return STATUS_SUCCESS;
+}
+
+//This routine represents a failing case for NoPagedCode
+_Use_decl_annotations_
+NTSTATUS
+DispatchRead (
+    PDEVICE_OBJECT DeviceObject,
+    PIRP Irp
+    )
+{  
+
+    KSPIN_LOCK  queueLock;
+    KIRQL oldIrql;
+    NTSTATUS status;
+
+    UNREFERENCED_PARAMETER(DeviceObject);
+    UNREFERENCED_PARAMETER(Irp);
+
+    KeInitializeSpinLock(&queueLock);
+     
+
+    KeAcquireSpinLock(&queueLock, &oldIrql);
+	
+    status = Irp->IoStatus.Status = STATUS_SUCCESS;
+    IoCompleteRequest(Irp, IO_NO_INCREMENT);
+    return STATUS_SUCCESS;
+}
+
+//This routine represents a failing case for NoPagingSegment.
+_Use_decl_annotations_
+NTSTATUS
+DispatchPower (
+    PDEVICE_OBJECT DeviceObject,
+    PIRP Irp
+    )
+{
+    NTSTATUS status = STATUS_PENDING;
+    PDRIVER_DEVICE_EXTENSION extension = (PDRIVER_DEVICE_EXTENSION)(DeviceObject->DeviceExtension); 
+    PAGED_CODE();
+    
+    
+    IoSetCompletionRoutine(Irp, CompletionRoutine, extension, TRUE, TRUE, TRUE);
+    
+    status = IoCallDriver(DeviceObject,Irp);
+    return status;
+}
+
+// MOVE THIS TO A SIMPLE FUNCTION CALL

```

Comment:
Is this comment a to-do for future work? If so please mark it with `TODO:`.  If not, what's the context of this comment?

Reply:
fixed